steam/guard.py

```diff
@@ -466,8 +496,14 @@ def validate_phone_number(self, phone_number):
         return resp
 
 
-class SteamAuthenticatorError(Exception):
-    pass
+class SteamAuthenticatorError(SteamError):
+    def __init__(self, message, eresult=None):
+        Exception.__init__(self, message, eresult)
+        self.message = message
+        self.eresult = EResult(eresult) if eresult is not None else None  #: :class:`.EResult`|:class:`None`
```

Comment:
We have plenty cases where there's no EResult available.

Reply:
All these is not needed. Simply inherit from `SteamError`. Don't set eresult to `None` where there is not result. Just leave the default.